buffalo/cmd/new.go

```diff
@@ -121,24 +136,38 @@ func notInGoPath(ag newapp.Generator) error {
 }
 
 func init() {
-	pwd, _ := os.Getwd()
-
-	app.App = meta.New(pwd)
-
 	decorate("new", newCmd)
 	RootCmd.AddCommand(newCmd)
-	newCmd.Flags().BoolVar(&app.AsAPI, "api", false, "skip all front-end code and configure for an API server")
-	newCmd.Flags().BoolVarP(&app.Force, "force", "f", false, "delete and remake if the app already exists")
-	newCmd.Flags().BoolVarP(&app.Verbose, "verbose", "v", false, "verbosely print out the go get commands")
-	newCmd.Flags().BoolVar(&app.SkipPop, "skip-pop", false, "skips adding pop/soda to your app")
-	newCmd.Flags().BoolVar(&app.WithDep, "with-dep", false, "adds github.com/golang/dep to your app")
-	newCmd.Flags().BoolVar(&app.SkipWebpack, "skip-webpack", false, "skips adding Webpack to your app")
-	newCmd.Flags().BoolVar(&app.SkipYarn, "skip-yarn", false, "use npm instead of yarn for frontend dependencies management")
-	newCmd.Flags().StringVar(&app.DBType, "db-type", "postgres", fmt.Sprintf("specify the type of database you want to use [%s]", strings.Join(pop.AvailableDialects, ", ")))
-	newCmd.Flags().StringVar(&app.Docker, "docker", "multi", "specify the type of Docker file to generate [none, multi, standard]")
-	newCmd.Flags().StringVar(&app.CIProvider, "ci-provider", "none", "specify the type of ci file you would like buffalo to generate [none, travis, gitlab-ci]")
-	newCmd.Flags().StringVar(&app.VCS, "vcs", "git", "specify the Version control system you would like to use [none, git, bzr]")
-	newCmd.Flags().IntVar(&app.Bootstrap, "bootstrap", app.Bootstrap, "specify version for Bootstrap [3, 4]")
+	newCmd.Flags().Bool("api", false, "skip all front-end code and configure for an API server")
+	newCmd.Flags().BoolP("force", "f", false, "delete and remake if the app already exists")
+	newCmd.Flags().BoolP("verbose", "v", false, "verbosely print out the go get commands")
+	newCmd.Flags().Bool("skip-pop", false, "skips adding pop/soda to your app")
+	newCmd.Flags().Bool("with-dep", false, "adds github.com/golang/dep to your app")
+	newCmd.Flags().Bool("skip-webpack", false, "skips adding Webpack to your app")
+	newCmd.Flags().Bool("skip-yarn", false, "use npm instead of yarn for frontend dependencies management")
+	newCmd.Flags().String("db-type", "postgres", fmt.Sprintf("specify the type of database you want to use [%s]", strings.Join(pop.AvailableDialects, ", ")))
+	newCmd.Flags().String("docker", "multi", "specify the type of Docker file to generate [none, multi, standard]")
+	newCmd.Flags().String("ci-provider", "none", "specify the type of ci file you would like buffalo to generate [none, travis, gitlab-ci]")
+	newCmd.Flags().String("vcs", "git", "specify the Version control system you would like to use [none, git, bzr]")
+	newCmd.Flags().Int("bootstrap", 3, "specify version for Bootstrap [3, 4]")
+	viper.BindPFlags(newCmd.Flags())
+	cobra.OnInitialize(initConfig)
+	newCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.buffalo.yaml)")
```

Comment:
I dislike this -- why not add a sane set of defaults and place the config in the buffalo project dir?

Reply:
This is a cobra convention:
https://github.com/spf13/cobra/blob/master/cobra/cmd/init.go#L183

Reply:
I suppose maybe we can use `$HOME/.buffalo/.buffalo.yml`??

Reply:
@robbyoconnor I used `$HOME/.buffalo.yaml ` because it was the default generated but it can be changed easily. If I'm not wrong @markbates mentioned it to an issue to be there.

Reply:
@robbyoconnor this is used with the buffalo new command, so it can’t live in the app, and needs to live somewhere “common”


Cobra, by convention, looks in the home directory for those dot files. Rails does, as well as a lot of similar projects, so I think that follows the path of least surprise. 

A loose feature “spec”:
By default buffalo new should read that file and use its defaults, if any. 

If no file, it should use the defaults in the bin. 

There should be a flag that lets you select a different config file other than the one at home. 

If the user passes a flag, it overrides anything in the config or in the bin 

Finally there should be a way of disabling the config file if the user doesn’t want to use it for a specific app

Reply:
The first 4 "specs" are "respected". As for the last maybe not so obvious you can pass a non existing file or a file of a different format that cobra can't read to the `--config` flag

Reply:
if someone passes a non-existent or badly formatted file it should error, not ignore the workflow. Why can't we add a --skip-config bool flag?

Reply:
Then I'll bow out here.